src/components/EmailNotifications/EmailNotificationsManager.js

```diff
@@ -0,0 +1,623 @@
+        ...subscriptionProgress,
+        previousScreen: screenId,
+      }))
+      setScreenId(DELETE_EMAIL_SCREEN)
+    }, [screenId])
+
+    const handleOnCancelDelete = useCallback(() => {
+      setScreenId(subscriptionProgress.previousScreen)
+    }, [subscriptionProgress.previousScreen])
+
+    const handleOnDelete = useCallback(async () => {
+      const { error, needsSignature } = await deleteJurorEmail(account)
+
+      if (error && !needsSignature) {
+        setSubscriptionProgress({ serviceError: true })
+        return
+      }
+
+      const deleteSettings = SETTINGS[DELETE_ACTION]
+
+      if (!needsSignature) {
+        if (isModal) {
+          setSubscriptionProgress(subscriptionProgress => ({
+            ...subscriptionProgress,
+            statusInfoTitle: deleteSettings.successInfo.title,
+            statusInfoText: `Your email ${subscriptionProgress.email} was succefully deleted. You can always re-subscribe from the notifications preferences later.`,
+          }))
+        }
+        setScreenId(
+          isModal ? SUCCESS_INFO_SCREEN : EMAIL_NOTIFICATIONS_FORM_SCREEN
+        )
+        return
+      }
+
+      setSubscriptionProgress(subscriptionProgress => ({
+        ...subscriptionProgress,
+        needSignature: needsSignature,
+        mode: isModal ? DELETE_ACTION : DELETE_ACTION_PREFERENCES,
+        signatureTitle: `Delete "${subscriptionProgress.email}"`,
+        signRequestText:
+          defaultSignRequestText + deleteSettings.signatureSettings.requestText,
+        signSuccessText: deleteSettings.signatureSettings.successText,
+      }))
+    }, [account, defaultSignRequestText, isModal])
+
+    const handleOnUnlockSettings = useCallback(() => {
+      const unlockSettings = SETTINGS[UNLOCK_SETTINGS_ACTION]
+
+      setSubscriptionProgress(subscriptionProgress => ({
+        ...subscriptionProgress,
+        needSignature: true,
+        mode: UNLOCK_SETTINGS_ACTION,
+        signatureTitle: unlockSettings.signatureSettings.title,
+        signRequestText:
+          defaultSignRequestText + unlockSettings.signatureSettings.requestText,
+        signSuccessText: unlockSettings.signatureSettings.successText,
+      }))
+    }, [defaultSignRequestText])
+
+    const handleOnUpdateEmail = useCallback(() => {
+      setScreenId(EMAIL_NOTIFICATIONS_FORM_SCREEN)
+    }, [])
+
+    const handleOnLockSettings = useCallback(async () => {
+      const { error, needsSignature } = await deleteCurrentSession(account)
+
+      if (error && !needsSignature) {
+        setSubscriptionProgress({ serviceError: true })
+        return
+      }
+
+      setScreenId(UNLOCK_NOTIFICATIONS_SCREEN)
+    }, [account])
+
+    const handleOnSignSuccess = useCallback((signHash, timestamp) => {
+      setSubscriptionProgress(subscriptionProgress => ({
+        ...subscriptionProgress,
+        startRequest: true,
+        signHash: signHash,
+        signTimestamp: timestamp,
+      }))
+    }, [])
+
+    useEffect(() => {
```

Comment:
I might not understand the full intent of this `useEffect()` and what interactions / API calls are meant to be driven.

From what I can tell, this is only launched from the `SignatureRequest` changing state (via the `handleOnSignSuccess()` callback). Once it is called, it:

1. Checks whether we have a session already
    - If not, creates one
2. Sends an arbitrary API request via `actions[subscriptionProgress.mode]`
3. If everything is successful and the success screen is requested, launches a timer to update the screen.

I have two questions, if what I've understood above is correct:

1. Do we need the `actions[subscriptionProgress.mode]` bit? Are we under the assumption that we could potentially push the `SignatureRequest` screen ahead of _any_ API interaction (because the session could have timed out and we need to ask the user to re-authenticate)?
1. Is the primarily goal of this provider (based on the assumption we'll turn this into a context provider) then to manage the `SignatureRequest` flow?

For the first question, I think this makes sense to control from here. I would just make sure we're making these calls _optimistically_, for example with deleting a juror:

1. We can assume that a user has already unlocked the notifications preferences (i.e. have a working session) before being able to access this functionality
2. If `handleOnDelete()` is called, _optimistically_ send a `DELETE` to `/users/<address>/email`
3. If the above fails due to an unauthenticated error, queue the delete action and set state to go back to the signature request screen
4. The modal/screen being driven shows the signature request screen and once validated, sends the delete action.